File: src/main/java/no/madsopheim/viatrumf/scraper/bff/NettbutikkController.java
package no.madsopheim.viatrumf.scraper.bff;

import no.madsopheim.viatrumf.scraper.bff.api.INettbutikkController;

import javax.inject.Inject;
import javax.inject.Singleton;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import java.util.List;

@Path("/nettbutikkar")
@Singleton
public class NettbutikkController implements INettbutikkController {

    @Inject
    FirestoreConnector firestoreConnector;

    @Override
    public List<String> getAlleNettbutikkar() {
        return firestoreConnector.finnAlleNettbutikkar();
    }

    @Override
    public List<Nettbutikk> hentInfoForNettbutikk(@PathParam("nettbutikk") String nettbutikk) {
        return firestoreConnector.query(nettbutikk);
    }
}

File: src/main/java/no/madsopheim/viatrumf/scraper/bff/NettbutikkController.kt
package no.madsopheim.viatrumf.scraper.bff

import jakarta.inject.Singleton
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.PathParam
import jakarta.ws.rs.Produces
import jakarta.ws.rs.core.MediaType

@Path("/nettbutikkar")
@Singleton
class NettbutikkController(val firestoreConnector: FirestoreConnector) {

    @Produces(MediaType.APPLICATION_JSON)
    @Path("/")
    @GET
    fun alleNettbutikkar(): List<String> = firestoreConnector.finnAlleNettbutikkar()

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    @Path("{nettbutikk}")
    fun hentInfoForNettbutikk(@PathParam("nettbutikk") nettbutikk: String): List<Nettbutikk> = firestoreConnector.query(nettbutikk)
}
